Tasks/AzureSpringCloudV0/deploymentProvider/azure-arm-spring-cloud.ts

```diff
@@ -0,0 +1,468 @@
+
+import tl = require('azure-pipelines-task-lib/task');
+import jsonPath = require('JSONPath');
+import webClient = require('azure-pipelines-tasks-azure-arm-rest-v2/webClient');
+import { AzureEndpoint } from 'azure-pipelines-tasks-azure-arm-rest-v2/azureModels';
+import { ServiceClient } from 'azure-pipelines-tasks-azure-arm-rest-v2/AzureServiceClient';
+import { ToError } from 'azure-pipelines-tasks-azure-arm-rest-v2/AzureServiceClientBase';
+import { uploadFileToSasUrl } from './azure-storage';
+import https = require('https');
+
+export const SourceType = {
+    JAR: "Jar",
+    SOURCE_DIRECTORY: "Source",
+    DOT_NET_CORE_ZIP: "NetCoreZip"
+}
+
+
+class UploadTarget {
+    private _sasUrl: string;
+    private _relativePath: string;
+
+    constructor(sasUrl: string, relativePath: string) {
+        this._sasUrl = sasUrl;
+        this._relativePath = relativePath;
+    }
+
+    public get sasUrl(): string {
+        return this._sasUrl;
+    }
+
+
+    public get relativePath(): string {
+        return this._relativePath;
+    }
+}
+
+const ASYNC_OPERATION_HEADER = 'azure-asyncoperation';
+
+export class AzureSpringCloud {
+
+    private _resourceId: string;
+    private _client: ServiceClient;
+
+
+    constructor(endpoint: AzureEndpoint, resourceId: string) {
+        tl.debug('Initializeing service client');
+        this._client = new ServiceClient(endpoint.applicationTokenCredentials, endpoint.subscriptionID, 30);
+        this._resourceId = resourceId;
+        tl.debug('Finished initializeing service client');
+    }
+
+
+    /**
+     * Encapsulates sending of Azure API requests.
+     * @param method 
+     * @param url 
+     * @param body 
+     */
+    protected sendRequest(method: string, url: string, body?: any): Promise<webClient.WebResponse> {
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = method;
+        httpRequest.uri = url;
+        if (body)
+            httpRequest.body = body;
+        tl.debug(`Sending ${method} request to ${url}`);
+        return this._client.beginRequest(httpRequest);
+    }
+
+
+    /* Parses the environment variable string in the form "-key value"
+     */
+    public static parseEnvironmentVariables(environmentVariables?: string): object {
+        if (!environmentVariables) {
+            return {};
+        }
+
+        var result = {};
+
+        var insideQuotes = false;
+        var curKey = '';
+        var curValue = '';
+        var readingKey = true;
+        for (var i = 0; i < environmentVariables.length; ++i) {
```

Comment:
This parsing can be error prone. (Complicated scenarios like - escaped quotes inside quotes)
Let's discuss this once.